pkg/table/contextimpl/table.go

```diff
@@ -73,3 +76,25 @@ func (ctx *TableContextImpl) GetMutateBuffers() *context.MutateBuffers {
 func (ctx *TableContextImpl) vars() *variable.SessionVars {
 	return ctx.Context.GetSessionVars()
 }
+
+// SetExtraIndexKeyPosInfo sets the extra index key pos info.
+func (ctx *TableContextImpl) SetExtraIndexKeyPosInfo(indexKeyPos map[int64][]int) {
+	ctx.hasExtraPosInfo = true
+	ctx.extraIndexKeyPosInfo = indexKeyPos
+}
+
+// GetExtraIndexKeyPosInfo gets the extra index key pos info.
+func (ctx *TableContextImpl) GetExtraIndexKeyPosInfo(idxID int64) []int {
+	return ctx.extraIndexKeyPosInfo[idxID]
+}
+
+// ResetExtraInfo resets the extra index key pos info.
+func (ctx *TableContextImpl) ResetExtraInfo() {
+	ctx.hasExtraPosInfo = false
+	ctx.extraIndexKeyPosInfo = nil
+}
+
+// HasExtraInfo returns whether the context has extra index key pos info.
+func (ctx *TableContextImpl) HasExtraInfo() bool {
+	return ctx.hasExtraPosInfo
```

Comment:
It's okay to do it. And later we'll support PartitionExpr, so it will become `len(extraIndexKeyPosInfo) == 0 && len(extraPartitionExpr) == 0` if we change to that way.